lib/src/main/java/graphql/nadel/engine/NadelIncrementalResultAccumulator.kt

```diff
@@ -115,6 +115,14 @@ class NadelIncrementalResultAccumulator(
             }
 
         if (readyAccumulators.isEmpty()) {
+            if(!hasNext) {
+                // We have to return hasNext=false to indicate to clients that there's no more data coming.
+                // Note: the spec allows an empty payload which only contains hastNext=false to be returned.
+                return DelayedIncrementalPartialResultImpl.newIncrementalExecutionResult()
+                    .incrementalItems(emptyList())
+                    .hasNext(false)
+                    .build()
```

Comment:
I think this belongs in `NadelIncrementalResultSupport`

This class just accumulates data. This is some incremental result logic.